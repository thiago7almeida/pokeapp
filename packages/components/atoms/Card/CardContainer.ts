import styled from 'styled-components/native';
import {responsive, spacing} from '@components';

export const CardContainer = styled.View<{backgroundColor: string}>`
  background-color: ${({backgroundColor}) => backgroundColor};
  border-radius: ${spacing.s16}px;
  padding: ${spacing.s8}px;
  flex-grow: 1;
  height: ${responsive.size(160, 'vertical')}px;
`;
