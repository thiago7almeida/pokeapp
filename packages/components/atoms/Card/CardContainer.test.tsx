import React from 'react';
import {render} from '@testing-library/react-native';
import '@testing-library/jest-native/extend-expect';

import {CardContainer} from './CardContainer';

jest.mock('@components', () => ({
  ...jest.requireActual('@components'),
  responsive: {
    size: (value: number) => value,
  },
  spacing: {
    s4: 4,
    s8: 8,
    s16: 16,
    s24: 24,
    s32: 32,
  },
}));

describe('CardContainer', () => {
  const backgroundColor = 'red';

  it('should render correctly', () => {
    const {toJSON} = render(
      <CardContainer backgroundColor={backgroundColor} />,
    );

    expect(toJSON()).toMatchSnapshot();
  });

  it('should have correct background color', () => {
    const {getByTestId} = render(
      <CardContainer
        backgroundColor={backgroundColor}
        testID="card-container"
      />,
    );

    expect(getByTestId('card-container')).toHaveStyle({backgroundColor});
  });

  it('should have correct border radius', () => {
    const {getByTestId} = render(
      <CardContainer
        backgroundColor={backgroundColor}
        testID="card-container"
      />,
    );

    expect(getByTestId('card-container')).toHaveStyle({borderRadius: 16});
  });

  it('should have correct padding', () => {
    const {getByTestId} = render(
      <CardContainer
        backgroundColor={backgroundColor}
        testID="card-container"
      />,
    );

    expect(getByTestId('card-container')).toHaveStyle({paddingTop: 8});
  });

  it('should have correct height', () => {
    const {getByTestId} = render(
      <CardContainer
        backgroundColor={backgroundColor}
        testID="card-container"
      />,
    );

    expect(getByTestId('card-container')).toHaveStyle({height: 160});
  });
});
