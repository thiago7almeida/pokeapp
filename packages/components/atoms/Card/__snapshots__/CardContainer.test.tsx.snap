// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CardContainer should render correctly 1`] = `
<View
  backgroundColor="red"
  style={
    [
      {
        "backgroundColor": "red",
        "borderRadius": 16,
        "flexGrow": 1,
        "height": 160,
        "paddingBottom": 8,
        "paddingLeft": 8,
        "paddingRight": 8,
        "paddingTop": 8,
      },
    ]
  }
/>
`;
